teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,465 @@
+---
+title: Tekton Enhancement Proposal Process
+authors:
+  - "@vdemeester"
+creation-date: 2020-03-10
+last-updated: 2020-04-23
+status: proposed
+---
+
+# Tekton Enhancement Proposal Process
+
+## Table of Contents
+
+<!-- toc -->
+**Table of Contents**
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Stewardship](#stewardship)
+- [Reference-level explanation](#reference-level-explanation)
+  - [What type of work should be tracked by a TEP](#what-type-of-work-should-be-tracked-by-a-tep)
+  - [TEP Template](#tep-template)
+  - [TEP Metadata](#tep-metadata)
+  - [TEP Workflow](#tep-workflow)
+  - [Git and GitHub Implementation](#git-and-github-implementation)
+  - [Prior Art](#prior-art)
+- [Examples](#examples)
+  - [Share Task and Pipeline as OCI artifact](#share-task-and-pipeline-as-oci-artifact)
+  - [PipelineResource re-design](#pipelineresource-re-design)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Unresolved Questions](#unresolved-questions)
+
+<!-- /toc -->
+
+
+## Summary
+
+A standardized development process for Tekton is proposed in order to
+
+- provide a common structure and clear checkpoints for proposing
+  changes to Tekton
+- ensure that the motivation for a change is clear
+- allow for the enumeration stability milestones and stability
+  graduation criteria
+- persist project information in a Version Control System (VCS) for
+  future Tekton users and contributors
+- support the creation of _high value user facing_ information such
+  as:
+  - an overall project development roadmap
+  - motivation for impactful user facing changes
+- ensure community participants are successfully able to drive changes
+  to completion across one or more releases while stakeholders are
+  adequately represented throughout the process
+
+This process is supported by a unit of work called a Tekton
+Enhancement Proposal (TEP). A TEP attempts to combine aspects of the
+following:
+
+- feature, and effort tracking document
+- a product requirements document
+- design document
+
+into one file which is created incrementally in collaboration with one
+or more [Working
+Groups](https://github.com/tektoncd/community/blob/master/working-groups.md)
+(WGs).
+
+This process does not block authors from doing early design docs using
+any means. It does not block authors from sharing those design docs
+with the community (during Working groups, on Slack, GitHub, …**.
+
+**This process acts as a requirement when a design docs is ready to be
+implemented or integrated in the `tektoncd` projects**. In other words,
+a change that impact other `tektoncd` projects or users cannot be
+merged if there is no `TEP` associated with it.
```

Comment:
i think we might need to define a bit better what "impact on other projects or users" means, since every change we make that's not refactoring or tests will affect users to some extent. (we can rule out most bugs maybe?)

Reply:
yep, we can rule out all the bugs really :wink: I'll add some examples.